### PR TITLE
Victor/eco 959/filtering out malformed offers

### PR DESCRIPTION
**Pull request by Doc999tor**
- [x] linked to a JIRA issue
- [x] description of the change
- [x] tests for the newly added code
- [x] if this is a breaking change, contains backward support code
- [x] if this requires client changes or partner updates, updated dependendants
- [x] were metrics and alerts added
- [x] should I change documentation

### CODE REVIEW

File: scripts/src/tests/helpers.ts [hunk 1]
```diff
@@ -180,6 +182,26 @@ export async function createOffers() {
...
+	const offerContents = (await OfferContent.findOne({ where: { name: offerName } }))!;
+	await offerContents.save(); // TypeORM parses the offers content json and saves it back to the table (it gets in not escaped) - such offer content is malformed and has to be filtered out in every .getAll
+
+	/*	sql for check whether the db has one malformed and wellformed offer content
+		```
+			select * from (select * from offer_contents where content like '{%' limit 1)
+			union all
+			select * from (select * from offer_contents order by id limit 1)
+		```
+	 */
```
Comment (doodyparizada) [inline scripts/src/tests/helpers.ts]: delete this

File: scripts/src/tests/helpers.ts [hunk 2]
```diff
@@ -180,6 +182,26 @@ export async function createOffers() {
 	}
 }
 
+async function createMalformedOffer(num: number) {
+	const offerName = generateId();
+	await createEarn(
+		`${ offerName }_earn${ num }`,
+		"GBOQY4LENMPZGBROR7PE5U3UXMK22OTUBCUISVEQ6XOQ2UDPLELIEC4J",
+		`earn${ num }`, `earn${ num }`, `earn${ num }`, `earn${ num }`, 100, 30, 1, `earn${ num }`, `earn${ num }`, "poll", animalPoll, ["ALL"]
```
Comment (doodyparizada) [inline scripts/src/tests/helpers.ts]: more line breaks

**Review by doodyparizada — CHANGES_REQUESTED** [review]

File: scripts/src/tests/helpers.ts (hunk 1 above)
Comment (Doc999tor) [reply scripts/src/tests/helpers.ts]: It's malformed offer checking script, I put it here in purpose
Delete it?

File: scripts/src/tests/helpers.ts (hunk 2 above)
Comment (Doc999tor) [reply scripts/src/tests/helpers.ts]: I copied it from the same function above
I'll add line breaks

Comment (doodyparizada) [reply scripts/src/tests/helpers.ts]: ok

File: scripts/src/tests/helpers.ts (hunk 1 above)
Comment (doodyparizada) [reply scripts/src/tests/helpers.ts]: Yes

File: scripts/src/tests/helpers.ts [hunk 3]
```diff
@@ -182,24 +181,40 @@ export async function createOffers() {
...
-			select * from (select * from offer_contents order by id limit 1)
-		```
-	 */
+	malformedOfferContent.content = "{a: 1}";
+	await malformedOfferContent.save();
+
+	return offer.id;
+}
+
+export async function createAppUser(offer: Offer, appId: string): Promise<User> {
```
Comment (doodyparizada) [inline scripts/src/tests/helpers.ts]: It's a weird name for what this function does once you removed it from the context. Maybe rename this createAppUserWithOffer

Comment (Doc999tor) [reply scripts/src/tests/helpers.ts]: I just moved existed inner function (closure) from another place
I'll change its name

**Review by doodyparizada — APPROVED** [review]

**Review by doodyparizada — APPROVED** [review]

**Pull request merged**